src/filter_tree/filter_tree.c

```diff
@@ -403,29 +404,46 @@ void _FilterTree_ApplyNegate(FT_FilterNode **root, uint negate_count) {
 	}
 }
 
+/* If a filter node that's not a child of a predicate is an expression,
+ * it should resolve to a boolean value. */
+static inline bool _FilterTree_ValidExpressionNode(const FT_FilterNode *root) {
+	bool valid = AR_EXP_ReturnsBoolean(root->exp.exp);
```

Comment:
It seems to me like we should evaluate the expression and test returned value,
what would happen is filters such as: `MATCH (a) WHERE a RETURN a` ?

Reply:
That's a good point - I think this function needs some work. The problem with evaluating the expression at this point is that we can have valid Record references in the function, as in:
```
MATCH (a) WHERE a.boolval RETURN a
MATCH (a) WHERE any(x IN a.arr WHERE x = 'accept') RETURN a
```
Which would error with:
```
(error) _AR_EXP_UpdateEntityIdx: No record was given to locate a value with alias a
```
If we tried to evaluate here.

`MATCH (a) WHERE a RETURN a` does (correctly) trigger an error here because we only naively accept function nodes, but that is not quite right either, since we could have queries like:
```
WITH true AS val WHERE val return val
CYPHER p=3 WITH 1 AS a WHERE $p RETURN a
```
The first of which should be accepted and the second should be rejected.

I'll give this area more thought!

Reply:
I've added run-time erroring to handle cases like those described here.